lib/app/features/gallery/providers/camera_provider.dart

```diff
@@ -1,47 +1,83 @@
 // SPDX-License-Identifier: ice License 1.0
 
 import 'package:camera/camera.dart';
+import 'package:ice/app/features/core/permissions/data/models/permissions_types.dart';
+import 'package:ice/app/features/core/permissions/providers/permissions_provider.dart';
 import 'package:ice/app/services/logger/logger.dart';
-import 'package:permission_handler/permission_handler.dart';
 import 'package:riverpod_annotation/riverpod_annotation.dart';
 
 part 'camera_provider.g.dart';
 
 @riverpod
-Future<Raw<CameraController?>> cameraController(CameraControllerRef ref) async {
-  final permissionStatus = await Permission.camera.status;
-  if (!permissionStatus.isGranted) {
-    final result = await Permission.camera.request();
-    if (!result.isGranted) {
-      Logger.log('Camera Permission denied');
+class CameraControllerNotifier extends _$CameraControllerNotifier {
+  CameraController? _cameraController;
+
+  @override
+  Future<Raw<CameraController>?> build() async {
+    ref.onDispose(() {
+      _cameraController?.removeListener(ref.notifyListeners);
```

Comment:
No need to remove listener, `dispose` does the thing.

Reply:
done